jans-cedarling/bindings/cedarling_python/src/config/bootstrap_config.rs

```diff
@@ -5,130 +5,468 @@
+/// :param id_token_trust_mode: Trust mode for ID tokens, either 'None' or 'Strict'.
+/// :param lock: Enables integration with Lock Master for policies and SSE events.
+/// :param lock_master_configuration_uri: URI where Cedarling can get JSON file with all required metadata about Lock Master, i.e. .well-known/lock-master-configuration.
+/// :param dynamic_configuration: Toggles listening for SSE config updates.
+/// :param lock_ssa_jwt: SSA for DCR in a Lock Master deployment. Cedarling will validate this SSA JWT prior to DCR.
+/// :param audit_log_interval: Interval (in seconds) for sending log messages to Lock Master (0 to disable).
+/// :param audit_health_interval: Interval (in seconds) for sending health updates to Lock Master (0 to disable).
+/// :param audit_health_telemetry_interval: Interval (in seconds) for sending telemetry updates to Lock Master (0 to disable).
+/// :param listen_sse: Toggles listening for updates from the Lock Server.
 ///
 /// Example
 /// -------
-/// ```
-/// from cedarling import BootstrapConfig, AuthzConfig, MemoryLogConfig, PolicyStoreConfig
-///
-/// # Create a BootstrapConfig with memory logging
-/// authz = AuthzConfig(application_name="MyApp")
-/// log_config = MemoryLogConfig(log_ttl=300)
-/// policy_store = PolicyStoreConfig(source=PolicyStoreSource(json='{...}'))
-/// jwt_config = JwtConfig(enabled=False)
+/// ```python
+/// from cedarling import BootstrapConfig
 ///
-/// bootstrap_config = BootstrapConfig(application_name="MyApp",authz_config=authz, log_config=log_config, policy_store_config=policy_store, jwt_config=jwt_config)
+/// # Example configuration
+/// bootstrap_config = NewBootstrapConfig(
+///     application_name="MyApp",
+///     policy_store_uri=None,
+///     policy_store_id="policy123",
+///     log_type="memory",
+///     log_ttl=60,
+///     user_authz=True,
+///     workload_authz=True,
+///     usr_workload_bool_op="AND",
+///     local_jwks="./path/to/your_jwks.json",
+///     local_policy_store=None,
+///     policy_store_local_fn="./path/to/your_policy_store.json",
+///     jwt_sig_validation=True,
+///     jwt_status_validation=False,
+///     at_iss_validation=True,
+///     at_jti_validation=True,
+///     at_nbf_validation=False,
+///     idt_iss_validation=True,
+///     idt_sub_validation=True,
+///     idt_exp_validation=True,
+///     idt_iat_validation=True,
+///     idt_aud_validation=True,
+///     userinfo_iss_validation=True,
+///     userinfo_sub_validation=True,
+///     userinfo_aud_validation=True,
+///     userinfo_exp_validation=True,
+///     id_token_trust_mode="Strict",
+///     lock=True,
+///     lock_master_configuration_uri=None,
+///     dynamic_configuration=False,
+///     lock_ssa_jwt=None,
+///     audit_log_interval=0,
+///     audit_health_interval=0,
+///     audit_health_telemetry_interval=0,
+///     listen_sse=False,
+/// )
 /// ```
 #[derive(Debug, Clone)]
 #[pyclass]
+#[pyo3(get_all, set_all)]
 pub struct BootstrapConfig {
```

Comment:
Maybe better implement this kind of structure in `cedarlind` (core)? To allow use it in other bindings?

Reply:
we need to define a struct that's compatible for use in the python bindings. if we do that, the core module will have `pyo3` as a dependency as well.

unless you don't want to make the bootstrap properties editable via code anymore and just load it from JSON or environment variables?

Reply:
Let's fix it in the future, because we have demo soon